django/contrib/auth/templatetags/auth.py

```diff
@@ -0,0 +1,28 @@
+from django.contrib.auth.hashers import UNUSABLE_PASSWORD_PREFIX, identify_hasher
+from django.template import Context, Library, Template
+from django.utils.html import format_html
+from django.utils.translation import gettext
+
+register = Library()
+
+
+@register.simple_tag
+def render_password_as_hash(value):
+    if not value or value.startswith(UNUSABLE_PASSWORD_PREFIX):
+        return format_html("<p><strong>{}</strong></p>", gettext("No password set."))
+    try:
+        hasher = identify_hasher(value)
+    except ValueError:
+        return format_html(
+            "<p><strong>{}</strong></p>",
+            gettext("Invalid password format or unknown hashing algorithm."),
+        )
+
+    hashed_summary = hasher.safe_summary(value)
+    items = [(gettext(key), value_) for key, value_ in hashed_summary.items()]
+
+    tpl = Template(
+        "<p>{% for key, value in items %} "
+        "<strong>{{ key }}</strong>: <bdi>{{ value }}</bdi>{% endfor %}</p>"
+    )
+    return tpl.render(Context({"items": items}))
```

Comment:
Thought: Using a template feels like a lot of overhead for what seems to be a `for` loop. Would `format_html_join` do the trick instead? Perhaps:

```python
return format_html(
    "<p>{}</p>",
    format_html_join("", "<strong>{}</strong>: <bdi>{}</bdi>", items),
)
```

With that said, it looks like the old template had a check for `if value` - should that be kept?

Reply:
> With that said, it looks like the old template had a check for if value - should that be kept?

It's a good question. Looking into `safe_summary()`, I don't think so. Seems to me to always have a value :thinking: 
At the very least it's untested and seems to have always been here b52c73008a9d67e9ddbb841872dc15cdd3d6ee01

Reply:
Ahhhh... it's because I have covered this with the other cases :grin: 
It shouldn't be needed :+1: